egs/cnceleb/v1/local/choose_utts_to_combine.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Vijayaditya Peddinti
+#           2016  Johns Hopkins University (author: Daniel Povey)
+#           2019  Tsinghua University (author: Jiawen Kang)
+#
+# Copied from utils/data/internal/choose_utts_to_combine.py
```

Comment:
What, if any, is the difference between this file and the one in utils/data/internal/choose_utts_to_combine.py?

If the differences are small, then I think it would be better to add some options to the original script to support whatever functionality is lacking, rather than copying the whole file and modifying it.

Reply:
Thanks a lot.
The only difference between this script and the one in utils/ is that we changed the return value of the function GetUtteranceGroups(). 
For the one in utils/, it will combine utterances across different speakers to satisfy the
duration constraint. This will involve some errors in spk2utt. So here we just combine utterances from the same speaker to ensure the correctness of the combined spk2utt.

Reply:
I think you'll have to add some option that controls this. I'm not sure what the best name for this would be. Maybe call it --merge-within-speakers-only, and have it default to false? I can't think of a more succinct option name at the moment... as long as the "help" text describes the purpose of the option it should be OK. 

Then you will probably want pass an option to GetUtteranceGroups that toggles between the two ways of merging. If this becomes complicated, you might decide that it's cleaner to make another function, maybe call it GetUtteranceGroupsWithinSpeaker. Then call whichever function is appropriate based on the value of merge_within_speakers_only.

Reply:
Thanks a lot. 
I have modified the scripts 'choose_utts_to_combine.py' and 'combine_short_segments.sh' in utils/, and remove them in local/. 

Besides, I found there is an error in our previous experiment due to the 'choose_utts_to_combine.py', and I will rebuild the i-vector and x-vector egs and update the experimental results. 
I will feed it back in tomorrow.